fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -425,27 +422,21 @@ def cleanup() -> None:
                             param.grad.data = param.grad.data.to(dtype=param.dtype)
 
                     # Async reduce for this buffer, log the future
-                    dst_global_rank = OSS.get_global_rank(self.process_group, dst_rank)
-
                     self._work_handles.append(
                         Workhandle(
                             handle=dist.reduce(
-                                tensor=param.grad.data, dst=dst_global_rank, group=self.process_group, async_op=True
+                                tensor=param.grad.data,
+                                dst=self._local_to_global_rank[dst_rank],
+                                group=self.process_group,
+                                async_op=True,
                             ),
                             callback=cleanup,
                         )
                     )
-                    self._reduced_grads += 1
 
-                    # Opportunistically try to empty the queue
+                    # Opportunistically try to empty the queue, free memory
                     self._try_consume_work_handle()
 
-                    # If all the reduce operations have been called,
-                    # make sure that all the asynchronous calls have concluded before moving on
-                    # and execute the delayed actions (release gradients, unroll the buckets)
-                    if self._reduced_grads == self._reduced_grads_max:
```

Comment:
this was sometimes blocking, but not needed given the Variable queue callback, and makes the code for this callback from complicated to grok